### ISSUE TITLE
Exam rank 03 - get_next_line traceback may be blank

### ISSUE THREAD

**Issue by ASR26** [JCluzet/42_EXAM#73]
I submit my code and get a failure, but traces are empty

**Comment by JCluzet** [comment]
Hi @ASR26 

Is there an error displayed on the standard output?
You don't have a `traces/` folder? 
Can you send me your `get_next_line`, so I can try to redo the error on my side? 
Thank you!

**Comment by ASR26** [comment]
`#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>

#ifndef BUFFER_SIZE
# define BUFFER_SIZE = 10
#endif
int	ft_line_len(char *str)
{
	int	i;

	i = 0;
	while (str[i] && str[i] != '\n')
		i++;
	if (str[i] == '\n')
		i++;
	return (i);
}

int	ft_strlen(char *str)
{
	int	i;

	i = 0;
	while (str[i])
		i++;
	return (i);
}

void	*ft_memcpy(void *dst, void *src, unsigned int n)
{
	unsigned char	*tmp;
	unsigned char	*src_tmp;

	if (!dst && !src)
		return (0);
	tmp = (unsigned char *) dst;
	src_tmp = (unsigned char *) src;
	while (n > 0)
	{
		*tmp++ = *src_tmp++;
		n--;
	}
	return (dst);
}

char	*ft_strchr(char *str, int c)
{
	while (*str)
	{
		if(*str == c)
			return (str);
		else
			(*str)++;
	}
	if (*str == c)
		return (str);
	return (0);
}

char	*ft_strjoin(char *s1, char *s2)
{
	int		newlen;
	char	*newstr;
	int		i;
	int		j;

	newlen = ft_strlen(s1) + ft_strlen(s2);
	newstr = (char *)malloc(sizeof(char) * newlen + 1);
	if(!newstr)
		return (newstr);
	i = 0;
	j = 0;
	while (s1[i])
	{
		newstr[i] = s1[i];
		i++;
	}
	while (s2[j])
	{
		newstr[i] = s2[j];
		i++;
		j++;
	}
	newstr[i] = '\0';
	return (newstr);
}

void	ft_clean_memory(char **memory)
{
	char	*new_memo;
	int		len;
	int 	new_len;

	if (*memory == NULL)
	{
		*memory = malloc (sizeof(char));
		**memory = '\0';
		return ;
	}
	len = ft_line_len(*memory);
	new_len = ft_strlen(&(*memory)[len] + 1);
	new_memo = (char *) malloc (sizeof(char) * new_len + 1);
	ft_memcpy(new_memo, &(*memory)[len], new_len);
	free (*memory);
	*memory = new_memo;
}

int	ft_fill_memory(char **memory, int fd)
{
	char	buffer[BUFFER_SIZE + 1];
	int		ret;
	char	*tmp;

	ft_clean_memory(memory);
	ret = 1;
	while (!ft_strchr(*memory, '\n') && ret)
	{
		ret = read (fd, buffer, BUFFER_SIZE);
		if (ret < 1)
			return (ret);
		buffer[ret] = '\0';
		tmp = *memory;
		*memory = ft_strjoin(*memory, buffer);
		free (tmp);
	}
	return (ret);
}

char	*ft_get_line(char **memory)
{
	char	*line;
	int		len;
	int		i;

	len = ft_line_len(*memory);
	if (len == 0)
		return (NULL);
	line = (char *) malloc(sizeof(char) * len + 1);
	i = 0;
	while (i < len)
	{
		line[i] = (*memory)[i];
		i++;
	}
	line[i] = '\0';
	return (line);
}

char	*get_next_line(int fd)
{
	static char	*memory = NULL;
	char		*line;

	if (ft_fill_memory(&memory, fd) == -1)
	{
		free (memory);
		memory = NULL;
		return (NULL);
	}
	line = ft_get_line(&memory);
	if (!line)
	{
		free (memory);
		memory = NULL;
	}
	return (line);
}`

this is the code I'm submitting (it's probably incorrect), i get a trace file but it's empty

**Comment by qu-b** [comment]
I'm having the same problem... The trace file is always empty.

**Comment by JCluzet** [comment]
I looked for a few minutes, but I couldn't find it. 
I don't work much on 42_EXAM anymore, especially since the get_next_line tester is an old version and hasn't been updated.

But I confirm that if it says FAILURE, you are wrong. 
I added a line in the traceback when it is empty to indicate that there is an error but that the traceback is broken for the moment.

If someone can find a solution and can pull the request I would be extremely grateful.
If not I will try to fix it later by putting a new version.
